Assets/Scripts/StaffManagement/StaffManager.cs

```diff
@@ -0,0 +1,59 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class StaffManager : MonoBehaviour
+{
+    [SerializeField] private SpawnNewStaff _spawnNewStaff;
+
+    private int _nextEmployeeID = 0;
+
+    public int LastEmployeeID => _nextEmployeeID - 1;
+    public Dictionary<int, Employee> Employees { get; private set; }
+
+    private void Awake() => Employees = new Dictionary<int, Employee>();
+
+    private void AddEmployeeToDictionary(Employee employee)
+    {
+        Employees.Add(_nextEmployeeID, employee);
+
+        _nextEmployeeID++;
+    }
+
+    /// <summary>
+    /// This function is called when you want to hire a type of employee
+    /// </summary>
+    /// <param name="type">The type of employee you want to hire</param>
+    public void HireEmployee(EmployeeTypes.EmployeeType type)
+    {
+        _spawnNewStaff.InstantiateEmployee(type);
+
+        Employee employee = _spawnNewStaff.NewEmployeeObjectSpawned.GetComponent<Employee>();
```

Comment:
Any particular reason for storing the last spawned employee and retrieving it from the class instead of directly returning the spawned employee from the `instantiateEmployee()` function?